### ISSUE TITLE
Optimize struct unpacking

### ISSUE THREAD

**Issue by tannewt** [adafruit/Adafruit_CircuitPython_LIS3DH#7]
```python
x = struct.unpack('<h', data[0:2])[0]
y = struct.unpack('<h', data[2:4])[0]
z = struct.unpack('<h', data[4:6])[0]
```

Slicing data causes an extra allocation. Struct should be able to take the whole buffer and split out the values at once.

**Comment by mrmcwethy** [comment]
I have a fix to Adafruit_Circuitpython_LIS3DH issue #7 Optimize struct unpacking in my development env.  Can someone  assign it to me or allow me contributor rights?

**Comment by mrmcwethy** [comment]
This issue is fixed as of oct 4, 2017
